docs/decisions/0001-role-based-access-control.rst

```diff
@@ -0,0 +1,39 @@
+1. Role Based Access Control
+----------------------------
+
+Status
+------
+
+Accepted
+
+
+Context
+-------
+In OpenedX system, we have concept of Learner and Admin. We have developed features that depend on the definition of
+these roles, including different user experience for Learners, and access to products such as the Admin Dashboard for
+Admins. As we embark on adding features, we need a way to assign users to these different roles and to have different
+services within the OpenedX ecosystem enforce permissions related to these roles. We also need the ability to create new
+roles that map to different sets of permissions as product needs arise.
+
+
+Decision
+--------
+The underlying engineering problem that OpenedX system facing is the lack of system that provides robust authorization
+capabilities. So a decision has been made in accordance with Architecture council that Role Based Access Control (RBAC)
+is the way to go, where users can be granted one or more roles which are each mapped to a set of permissions. Using RBAC
+we can control granularity of access for a user. As a result of that decision, this codebase was created to provide all
+the tools and tricks to implement role based authorization in different parts of the OpenedX ecosystem.
```

Comment:
"As a result of that decision, this codebase was created to provide all the tools and tricks to implement role based authorization in different parts of the OpenedX ecosystem."
->
"To support RBAC in the OpenedX ecosystem, this repository was created to provide standardized ways of defining roles and permission mappings and performing permission checks in any django application that requires RBAC."